srml/support/src/storage/generator/linked_map.rs

```diff
@@ -360,4 +401,50 @@ where
 			Ok(len)
 		}
 	}
+
+	fn translate<K2, V2, TK, TV>(translate_key: TK, translate_val: TV) -> Result<(), ()>
+		where K2: FullCodec + Clone, V2: Decode, TK: Fn(K2) -> K, TV: Fn(V2) -> V
+	{
+		let head_key = read_head::<K2, G::KeyFormat>().ok_or(())?;
+		let prefix = G::prefix();
+
+		let mut current_key = head_key.clone();
+
+		let translate_linkage = |old: Linkage<K2>| -> Linkage<K> {
+			Linkage {
+				previous: old.previous.map(&translate_key),
+				next: old.next.map(&translate_key),
+			}
+		};
+
+		loop {
+			let old_raw_key = G::KeyFormat::storage_linked_map_final_key(prefix, &current_key);
+			let (val, linkage) = read_with_linkage::<K2, V2>(old_raw_key.as_ref()).ok_or(())?;
+			let next = linkage.next.clone();
+
+			let val = translate_val(val);
+			let linkage = translate_linkage(linkage);
+
+			// now that we've successfully migrated the storage, kill the old key.
+			unhashed::kill(old_raw_key.as_ref());
```

Comment:
You are removing the old one, without knowing if there will be an error.

Reply:
The only possible error afterwards is if the next iteration of the loop doesn't decode correctly with `K2` and `V2`. We've already decoded correctly as `K2` and `V2`, so there are only 2 ways that can happen:
  - `K2` and `V2` aren't actually the correct old format and only decoded before by chance.
  - The storage is in a malformed state already. We can't fix that.